discovery-provider/src/utils/prometheus_metric.py

```diff
@@ -58,16 +58,14 @@ def save_time(self, labels=None, start_time=None):
         self.save(self.elapsed(start_time), labels)
 
     def save(self, value, labels=None):
+        this_metric = self.metric
```

Comment:
Yup, it does! 

`self.metric.labels()` returns a new Metric object which we will then use to call `.set()` or `value()` with.

We don't seek to change `self.metric` since we might want to call `self.metric.set()` later without including any labels, but sometimes we do want to call `self.metric.labels(labels).set()`.

New PR here: https://github.com/AudiusProject/audius-protocol/pull/2901